scripts/run-spark.py

```diff
@@ -115,11 +116,19 @@ def get_spark_args(args):
 
 def run(cmd, **kwargs):
     logging.info("$ " + cmd)
-    # TODO: capture output while printing
-    return subprocess.run(
-        cmd, capture_output=True, shell=True, check=True, encoding="utf-8", **kwargs
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
     )
 
+    output_list = []
+    for c in iter(lambda: process.stdout.read(1), b""):
```

Comment:
is this only reading one byte at a time? curious if this will affect performance. I've also seen solutions that does `readline()`. But if it works let's use this for now.

Reply:
I don't believe this effects performance because Spark does not factor in logging times into the run (at least from what I've seen). Using `readline` is a little messy since I believe you may have to do error checking to see if you've actually read in a full line.